Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM golang:1.12-alpine as build_base
+WORKDIR /build
+ENV GO111MODULE=on
+RUN apk add --update git
+COPY go.mod .
+COPY go.sum .
+RUN go mod download
+
+FROM build_base as build
+RUN apk add --update make
+COPY . .
+RUN make build
+
+FROM alpine
+COPY --from=build /build/secrethub /usr/bin/secrethub
+RUN apk add --no-cache ca-certificates && \
+    update-ca-certificates
+ENTRYPOINT ["secrethub"]
```

Comment:
With `secrethub` as `ENTRYPOINT`, can you test how well extending it works? And do we want to support it that you can `docker exec` into it at runtime and run ad hoc commands in a shell?

Reply:
We're working on making it possible to run all commands without having to execute in shell form. For example, see https://github.com/secrethub/secrethub-cli/issues/37 for adding a `--template` flag to the `inject` command so that that can be run in exec form.

Note that in the meantime, you can override the entrypoint.

---

As for the docker exec: Do you mean to have an initial command that does not exit, so that `docker exec` can be called at arbitrary moments on the same container?

That might be useful in some cases. However, I don't think that should be the default. The image can be used like that by overriding the entrypoint.